ipaclient/install/ipa_client_samba.py

```diff
@@ -0,0 +1,726 @@
+    secrets_key = "SECRETS/MACHINE_LAST_CHANGE_TIME/{}".format(
+        domain["netbios_name"]
+    )
+    args = [paths.TDBTOOL, paths.SECRETS_TDB, "store", secrets_key, "2\\00"]
+    try:
+        ipautil.run(args)
+    except ipautil.CalledProcessError as e:
+        logger.error(
+            "Cannot prepare machine account creds in Samba. " "Error: %s",
+            str(e),
+        )
+        raise
+
+    secrets_key = "SECRETS/MACHINE_PASSWORD/{}".format(domain["netbios_name"])
+    args = [paths.TDBTOOL, paths.SECRETS_TDB, "store", secrets_key, "2\\00"]
+    try:
+        ipautil.run(args)
+    except ipautil.CalledProcessError as e:
+        logger.error(
+            "Cannot prepare machine account creds in Samba. " "Error: %s",
+            str(e),
+        )
+        raise
+
+    # Finally, set actual machine account's password
+    args = [paths.NET, "changesecretpw", "-f"]
+    try:
+        ipautil.run(args, stdin=password, encoding="utf-8")
+    except ipautil.CalledProcessError as e:
+        logger.error(
+            "Cannot set machine account creds in Samba. " "Error: %s", str(e)
+        )
+        raise
+
+
+def configure_default_groupmap(fstore, statestore, options, domain):
+    args = [
+        paths.NET,
+        "groupmap",
+        "add",
+        "sid=S-1-5-32-546",
+        "unixgroup=nobody",
+        "type=builtin",
+    ]
+
+    logger.info("Map BUILTIN\\Guests to a group 'nobody'")
+    try:
+        ipautil.run(args)
+    except ipautil.CalledProcessError as e:
+        if "already mapped to SID S-1-5-32-546" not in e.stdout:
+            logger.error(
+                'Cannot map BUILTIN\\Guests to a group "nobody". ' "Error: %s",
+                str(e),
+            )
+            raise
+
+
+def set_selinux_booleans(booleans, statestore, backup=True):
+    def default_backup_func(name, value):
+        statestore.backup_state("selinux", name, value)
+
+    backup_func = default_backup_func if backup else None
+    try:
+        tasks.set_selinux_booleans(booleans, backup_func=backup_func)
+    except SetseboolError as e:
+        print("WARNING: " + str(e))
+        logger.info("WARNING: %s", str(e))
+
+
+def harden_configuration(fstore, statestore, options, domain):
+    # Add default homes share so that users can log into Samba
+    if not options.no_homes:
+        set_selinux_booleans(
+            constants.SELINUX_BOOLEAN_SMBSERVICE["share_home_dirs"], statestore
+        )
+    # Allow Samba to access NFS-shared content
+    if not options.no_nfs:
+        set_selinux_booleans(
+            constants.SELINUX_BOOLEAN_SMBSERVICE["reshare_nfs_with_samba"],
+            statestore,
+        )
+
+
+def uninstall(fstore, statestore, options):
+    # Shut down Samba services and disable them
+    smb = services.service("smb", api)
+    winbind = services.service("winbind", api)
+    for svc in (smb, winbind):
+        if svc.is_running():
+            svc.stop()
+        svc.disable()
```

Comment:
Should we disable services we don't enable?

Reply:
We should because we removed the configuration.